Framework/Algorithms/src/RemoveExpDecay.cpp

```diff
@@ -100,15 +105,23 @@ void MuonRemoveExpDecay::exec() {
       throw std::invalid_argument(
           "Spectra size greater than the number of spectra!");
     }
-
-    // Remove decay from Y and E
-    outputWS->setHistogram(specNum, removeDecay(inputWS->histogram(specNum)));
-
-    // do scaling and subtract 1
-    const double normConst = calNormalisationConst(outputWS, spectra[i]);
-    outputWS->mutableY(specNum) /= normConst;
-    outputWS->mutableY(specNum) -= 1.0;
-    outputWS->mutableE(specNum) /= normConst;
+    auto fsdaf = (inputWS->y(specNum));
+    auto emptySpectrum = std::all_of(inputWS->y(specNum).begin(),
+                                     inputWS->y(specNum).end(), isZero);
+    if (emptySpectrum) {
+      // if the y values are all zero do not change them
+      m_log.warning("waaa " + std::to_string(specNum));
```

Comment:
I presume you've left this in by accident? :stuck_out_tongue: Might something like `Dead detector encountered at spectrum number " + blah` be more informative?

Reply:
I agree